### PR TITLE
using fwd links for specifying winrm utilities

### PR DESCRIPTION
**Pull request by arjgupta**

### CODE REVIEW

**Comment by arjgupta** [comment]
for the azure file copy task

File: Tasks/AzureFileCopy/Utility.ps1 [hunk 1]
```diff
@@ -1275,6 +1275,17 @@ function Is-WinRMCustomScriptExtensionExists
     $isExtensionExists
 }
 
+function Get-TargetUriFromFwdLink {
+    param(
+        [string]$fwdLink
+    )   
+
+    Write-Verbose "Trying to get the target uri from the fwdLink: $fwdLink"
+    $targetUri =  $(Invoke-WebRequest $fwdLink).BaseResponse.Uri.AbsoluteUri
```
Comment (kmkumaran) [inline Tasks/AzureFileCopy/Utility.ps1]: I see few examples where it is mentioned that it is safe to be explicit setting this request.AllowAutoRedirect = true; when we are intending to find the redirect Url.

Comment (kmkumaran) [reply Tasks/AzureFileCopy/Utility.ps1]: This was not proxy aware earlier. Would it be better to fix that as well?

File: Tasks/AzureFileCopy/make.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
         "nugetv2": [
             {
                 "name": "VstsTaskSdk",
-                "version": "0.8.2",
+                "version": "0.10.0",
```
Comment (kmkumaran) [inline Tasks/AzureFileCopy/make.json]: Make sure our CDP is green post this.

File: Tasks/AzureFileCopy/Utility.ps1 [hunk 3]
```diff
@@ -1275,6 +1275,24 @@ function Is-WinRMCustomScriptExtensionExists
...
+    Write-Verbose "Trying to get the target uri from the fwdLink: $fwdLink"
+    $proxy = Get-VstsWebProxy
+    Add-Type -AssemblyName System.Net.Http
+    $HttpClientHandler = New-Object System.Net.Http.HttpClientHandler
+    $HttpClientHandler.Proxy = $proxy
+    $HttpClientHandler.AllowAutoRedirect = $false
+    $HttpClient = New-Object System.Net.Http.HttpClient -ArgumentList $HttpClientHandler
+    $response = $HttpClient.GetAsync($fwdLink)
+    $response.Wait()
+    $targetUri =  $response.Result.Headers.Location.AbsoluteUri
```
Comment (kmkumaran) [inline Tasks/AzureFileCopy/Utility.ps1]: can you assert for null here?

Comment (arjgupta) [reply Tasks/AzureFileCopy/Utility.ps1]: done

File: Tasks/AzureFileCopy/Utility.ps1 [hunk 4]
```diff
@@ -1310,8 +1321,8 @@ function Add-AzureVMCustomScriptExtension
           [string]$location,
           [string]$connectedServiceName)
 
-    $configWinRMScriptFile="https://raw.githubusercontent.com/Azure/azure-quickstart-templates/501dc7d24537e820df7c80bce51aba9674233b2b/201-vm-winrm-windows/ConfigureWinRM.ps1"
-    $makeCertFile="https://raw.githubusercontent.com/Azure/azure-quickstart-templates/501dc7d24537e820df7c80bce51aba9674233b2b/201-vm-winrm-windows/makecert.exe"
+    $configWinRMScriptFileFwdLink ="https://aka.ms/vstsconfigurewinrm"
```
[the remark this answers is not in the document]
Comment (arjgupta) [reply Tasks/AzureFileCopy/Utility.ps1]: done

**Pull request merged**